doozerlib/rpmcfg.py

```diff
@@ -384,52 +397,53 @@ def build_rpm(self, version, release, terminate_event, scratch=False, retries=3,
         """
         if local:
             raise DoozerFatalError("Local RPM build is not currently supported.")
-        self.set_nvr(version, release)
-        self.tito_setup()
-        self.update_spec()
-        self.commit_changes(scratch)
-        action = "build_rpm"
-        record = {
-            "specfile": self.specfile,
-            "source_head": self.source_head,
-            "distgit_key": self.distgit_key,
-            "rpm": self.rpm_name,
-            "version": self.version,
-            "release": self.release,
-            "message": "Unknown failure",
-            "status": -1,
-            # Status defaults to failure until explicitly set by succcess. This handles raised exceptions.
-        }
 
-        try:
-            def wait(n):
-                self.logger.info("Async error in rpm build thread [attempt #{}]: {}".format(n + 1, self.qualified_name))
-                # Brew does not handle an immediate retry correctly, wait
-                # before trying another build, terminating if interrupted.
-                if terminate_event.wait(timeout=5 * 60):
-                    raise KeyboardInterrupt()
+        with self.runtime.get_dir_lock(self.source_path):
```

Comment:
don't think we have an RPMs building from the same source, only images. still, i suppose it could theoretically happen.